src/mod/codecs/mod_amrwb/mod_amrwb.c

```diff
@@ -144,7 +144,11 @@ static switch_bool_t switch_amrwb_info(unsigned char *encoded_buf, int encoded_d
 		tocs = encoded_buf;
 		index = (tocs[0] >> 3) & 0x0f;
 		if (index > SWITCH_AMRWB_MODES) {
-			switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "AMRWB decoder (OA): Invalid TOC 0x%x\n", index);
+			if (!codec->session) {
+				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "AMRWB decoder (OA): Invalid TOC 0x%x\n", index);
+			} else {
+				switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(codec->session), SWITCH_LOG_ERROR, "AMRWB decoder (OA): Invalid TOC 0x%x\n", index);
```

Comment:
SWITCH_CHANNEL_SESSION_LOG will check if codec->session is NULL for you

Reply:
but will it log with "SWITCH_CHANNEL_LOG" if session is NULL ?

Reply:
yes